pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2864,6 +2864,11 @@ type StorageScopeSpec struct {
 	// Whether to allow updating the device class after the OSD is initially provisioned
 	// +optional
 	AllowDeviceClassUpdate bool `json:"allowDeviceClassUpdate,omitempty"`
+	// Whether Rook will resize the OSD CRUSH weight when the OSD PVC size is increased.
+	// This allows cluster data to be rebalanced to make most effective use of new OSD space; however, data rebalancing can cause temporary cluster slowdown.
+	// +optional
+	// +kubebuilder:default=true
```

Comment:
In my experience, kubebuilder defaults that don't match the default data type have strange behaviors, like the CR originally parsing with `false`, then immediately being "updated" to `true`, sometimes causing a double reconcile.

1. @travisn do we really want the default to be true?
2. If so, then we should consider changing this to instead be `disallowOsdCrushWeightUpdate` instead with a false default.

Reply:
Agreed, setting the kubebuilder default has caused issues in the past so let's avoid it. I'd suggest it's ok to keep the default as false and not rename the var. It would be nice to keep the naming consistent with `allowDeviceClassUpdate`. Since the example will have it set to `true`, most users who care will probably have it set anyway, and it's easy to change to true for the users who realize it later.

Reply:
@travisn If that the case then we need to set this value from ocs operator to true, 
Do we really wanna add that complex behavior, 
Either way downstream it should be always true.

Reply:
I see at other places in api we also set default to `true`, but if it really has bugs we can opt out for this instance

Reply:
Anway what I understand is this flag is provided for those, who really don't want to work how ceph re-wieight works, and add their custom weights.

So if we think for default way it should be really how should ceph works for default.
And another thing we are doing is not re-weighting for the cases which cant be true like new weight less than the previous weight.

Reply:
I think the default for upstream actually makes sense as `false`. Upstream users may be setting custom weights, and could be surprised by rook starting to update them automatically. It may not be likely, but it does feel like the better upstream approach. Then we don't need to worry about if there is still an issue with the `default=true` in the CRD.